gateway/assets/script/bootstrap.js

```diff
@@ -180,29 +180,32 @@ app.controller("home", ['$scope', '$log', '$http', '$location', '$interval', '$f
                 });
         };
 
-        var refreshData = function() {
+        var refreshData = function () {
             var previous = $scope.functions;
 
-            var cl = function(previousItems) {
-                $http.get("../system/functions").then(function(response) {
-                    if (response && response.data) {
-                        if (previousItems.length != response.data.length) {
-                            $scope.functions = response.data;
-                            
-                            // update the selected function object because the newly fetched object from the API becomes a different object
-                            var filteredSelectedFunction = $filter('filter')($scope.functions, {name: $scope.selectedFunction.name}, true);
-                            if (filteredSelectedFunction && filteredSelectedFunction.length > 0) {
-                                $scope.selectedFunction = filteredSelectedFunction[0];
-                            } else {
-                                $scope.selectedFunction = undefined;
-                            }
+            var cl = function (previousItems) {
+                $http.get("../system/functions").then(function (response) {
+                    if (response == null || response.data == null) {
```

Comment:
I don't think we should change this because this may not catch when the `response` or `response.data` is `undefined`. Also, we should always use triple equals `===` whenever possible (which some existing code doesn't include).

Reply:
I am familiar with the `===` rule in JS, but explicitly chose `==` because it does catch `undefined`.

Reply:
Thanks for clearing it out! I wasn't sure how loose it was. 

```
response && response.data
```

The previous statement is probably an anti-pattern in JS if we think strictly, as it catches falsy values (like 0, "").
(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else#Description)

IMHO, I think the previous code was more readable, but am not against this change either.

Reply:
I am ok with rewriting it, but my reason for writing it this way it reduces the nesting / indentation level and keeps the primary logic close to the primary level of the. I generally prefer to check for errors/issues and return early and allow the expected or happy path to be the primary logic of a method

Reply:
Oh, sorry Lucas, I was missing the whole point. Somehow I only thought the code was changed from no `null` statement to using `null` in the if statement. So I thought the change after that was due to formatting. Sorry about me not understanding the intent.

> I generally prefer to check for errors/issues and return early

This cleared what I thought was strange and I agree about returning early.

Reply:
You can do a check with `(!response && !response.data)` as well.

Reply:
I prefer the original code for this which like Ken says is more readable. Also unless I'm reading this wrong, we reversed the logic. Can you help me understand the change?

```
// do something if there is a value for response.data
if (response && response.data) {
```

Changed to:

```
// do something if and only if either are null?

if (response == null || response.data == null) {
```

Reply:
if there is no response or there is no data, then stop processing the response and return immediately.  Ideally, this would actually throw and error or set an error state in the UI, but that is beyond the scope of this PR

I did this so that

1) the primary intention of the function (update the function data) is the main scope of the function, error and edge cases are checked and dealt with prior to the success case; and
2) it reduces the number of indented blocks and the indentation level as a whole, i find long nested chunks of code hard to read